lint/elseif_checker.go

```diff
@@ -30,15 +32,25 @@ if cond1 {
 }`
 }
 
+func (c *elseifChecker) Init() {
+	c.skipBalanced = c.ctx.params.Bool("skipBalanced", true)
+}
+
 func (c *elseifChecker) VisitStmt(stmt ast.Stmt) {
 	if stmt, ok := stmt.(*ast.IfStmt); ok {
 		elseBody, ok := stmt.Else.(*ast.BlockStmt)
 		if !ok || len(elseBody.List) != 1 {
 			return
 		}
-		if astp.IsIfStmt(elseBody.List[0]) {
-			c.warn(stmt.Else)
+		if !astp.IsIfStmt(elseBody.List[0]) {
+			return
+		}
+		balanced := len(stmt.Body.List) == 1 &&
+			astp.IsIfStmt(stmt.Body.List[0])
+		if balanced && c.skipBalanced {
```

Comment:
What about keeping everything together?
```
balanced := c.skipBalanced && 
	len(stmt.Body.List) == 1 &&
	astp.IsIfStmt(stmt.Body.List[0])
if balanced {
```

Reply:
This is not `balanced` condition if you include everything in it.

Reply:
Ah, I see, you suggesting to move `c.skipBalanced` to `balanced` expression.
I don't think it improves readability, actually.